### DISCUSSION TITLE
单机版2.2.9 docker 部署，磁盘占用多

### DISCUSSION THREAD

**Discussion by hemin110** [milvus-io/milvus#24886] [Q&A and General discussion] [answered]
minio 磁盘占用异常， 一共存储123495 条1024维的向量   V2.2.9
[root@localhost volumes]# ls
etcd  milvus  minio
[root@localhost volumes]# du -sh *
537M    etcd
637M    milvus
330G    minio

[root@localhost files]# ls
index_files  insert_log  stats_log
[root@localhost files]# du -sh *
534M    index_files
329G    insert_log
303M    stats_log

**Comment by xiaofan-luan** [comment]
pls offer us logs for datacoord, indexcoord and querycoord so we can figureout the reason why.

**Reply by xiaofan-luan** [reply]
Did your upgrade from any old version or start from 2.2.9?

**Reply by hemin110** [reply]
不是升级的，直接使用的2.2.9 如何取出相关日志，直接用docker logs 导出的可以吗？

**Reply by yanliang567** [reply]
For Milvus installed with docker-compose, you can use docker-compose logs > milvus.log to export the logs.

**Reply by yanliang567** [reply]
did you have any delete operations?

**Reply by hemin110** [reply]
测试的时候执行过删除

**Reply by yanliang567** [reply]
how many data did you delete during testing? one more question is did you remove the volume folder every time you deploy a new milvus instance? I am asking because it could be some dirty data in the volume folder.

**Reply by hemin110** [reply]
测试的时候每次删除78条数据，可能是删除了4-5次， 移除卷后是重新部署的，我们是需要把卷直接发给另外一个人进行复用

**Comment by hemin110** [comment]
这是相关的日志
链接：https://pan.baidu.com/s/1btkWRxooFotUSfhUCm_VuA 
提取码：kood

**Comment by ruixiu** [comment]
我使用的也是2.2.9单机版，数据盘是单独挂载的一个磁盘130多万的2048维向量，存储占用如下（应该是比较正常的）：
![image](https://github.com/milvus-io/milvus/assets/9940354/b9fc3fda-7951-4400-9244-034b75c36d94)
不过buff/cache增长是真快，一直没找到是什么原因，并且系统盘也在持续增长，也找不到到底是哪个文件占用了（重启milvus容器就释放，很怪）
![image](https://github.com/milvus-io/milvus/assets/9940354/bf7df04f-5ee5-4bdf-bc7f-ba3525a867ec)

**Reply by yanliang567** [reply]
if you are talking about the runtime log here:
- Docker purges the runtime log into /var/lib/docker/containers/[container_id]/

please try to update the config:
 ```
 "log-driver": "json-file",
  "log-opts": {
    "max-size": "128m",
    "max-file": "3" 
  }
```

The related topic: https://docs.docker.com/config/containers/logging/json-file/

**Comment by jiaoew1991** [comment]
According to your description here https://github.com/milvus-io/milvus/discussions/24240#discussioncomment-6171582, the frequency of flush is too high.

**Reply by hemin110** [reply]
how to release disk and how to control flush frequency

**Reply by jiaoew1991** [reply, marked as answer]
It is not recommended to call the flush API as the system will automatically flush.

Expired files will be automatically cleared and it takes 24 hours.

**Reply by hemin110** [reply]
thanks
